### DISCUSSION TITLE
How do I disable MapLibre-managed attribution completely before re-implementing it myself?

### DISCUSSION THREAD

**Discussion by eslksw** [maplibre/maplibre-gl-js#5008] [Q&A] [answered]
So, I have a relatively specific qualm: I am developing an app using MapLibre. The app's UI is decently complex, so I would like to handle the positioning (and rendering in general) of the attribution myself using a custom React component. However, I can't seem to be able to remove it completely. Anything I could find on the topic concerns re-positioning using AttributionControl, but I need something more fine-grained than that. I tried zapping the DOM element using good old `remove()` and while it works, it feels a bit too hacky. Any hints on how to achieve this? Thank you in advance!

**Comment by HarelM** [comment]
I think you can set attribution to `false` when initializing the map, can't you?

**Reply by eslksw** [reply]
Apparently not, since I get the following error if I try:

`Error: sources.osm.attribution: string expected, boolean found`

**Reply by HarelM** [reply]
Please share a reproduction.

**Reply by HarelM** [reply, marked as answer]
See map options' attribution control:
https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/MapOptions/

**Reply by eslksw** [reply]
Sure!

``` javascript
const style = {
    version: 8,
    sources: {
      osm: {
        type: "raster",
        tiles: ["https://a.tile.openstreetmap.org/{z}/{x}/{y}.png"],
        tileSize: 256,
        attribution: false // <- this throws the error in question
      },
    },
    layers: [
      {
        id: "osm",
        type: "raster",
        source: "osm",
      },
    ],
  };

useEffect(() => {
    if (map) return;

    const newmap = new maplibregl.Map({
      container: mapContainerRef.current,
      style: style,
      center: [lng, lat],
      minZoom: 2,
      maxZoom: 10,
      zoom: zoom,
      attribution: false,  // <- this doesn't do anything
    });

    setMap(newmap);
  }, [map, lng, lat, zoom, style]);
```

**Reply by eslksw** [reply]
> See map options' attribution control: https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/MapOptions/

Yes, that was it, thank you!

**Reply by HarelM** [reply]
I would advise to use typescript, it would have saved you the grief...
